dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpTransport.java

```diff
@@ -0,0 +1,211 @@
+        return mcpRequest.getParams(type, jsonb);
+    }
+
+    /**
+     * Sends the MCP Server's response with the results for the client's request
+     *
+     * @param result the result field to include in the MCP Server response
+     * @throws IOException
+     */
+    public void sendResponse(Object result) {
+        McpResponse mcpResponse = new McpResultResponse(mcpRequest.id(), result);
+        jsonb.toJson(mcpResponse, writer);
+    }
+
+    /**
+     * Sends a 202 response for requests that don't expect anything to be sent back
+     *
+     * E.g. when a MCP client sends a notification, it'a not expecting anything back in the response
+     */
+    public void sendEmptyResponse() {
+        res.setStatus(HttpServletResponse.SC_ACCEPTED);
+    }
+
+    /**
+     * sends custom error message response back to client depending on the error
+     *
+     * @param e the exception that was caught
+     * @throws IOException
+     */
+    public void sendError(Exception e) throws IOException {
+        Tr.error(tc, "Unexpected Server Error. Method={0} RequestURI={1} RequestQuery={2}", req.getMethod(), req.getRequestURI(), req.getQueryString());
+        res.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "An unexpected error occurred. Please try again later.");
+    }
+
+    /**
+     * This method is responsible for sending a JSON-RPC error response.
+     * It takes a JSONRPCException as an argument and constructs an McpErrorResponse object.
+     * This error response is then serialised to JSON and written to the provided writer.
+     *
+     * @param e The JSONRPCException to be included in the error response.
+     */
+    public void sendJsonRpcException(JSONRPCException e) {
+        McpResponse mcpResponse = new McpErrorResponse(mcpRequest == null ? "" : mcpRequest.id(), e);
+        jsonb.toJson(mcpResponse, writer);
+    }
+
+    /**
+     * Sends a response back to client when an HTTP exception occurs
+     *
+     * @param e The HttpResponseException to be included in the error response.
+     * @throws IOException
+     */
+    public void sendHttpException(HttpResponseException e) throws IOException {
+        switch (e.getStatusCode()) {
+            case (HttpServletResponse.SC_NOT_ACCEPTABLE) -> {
+                res.setStatus(HttpServletResponse.SC_NOT_ACCEPTABLE);
+                res.setContentType("application/json");
+            }
+            case (HttpServletResponse.SC_BAD_REQUEST) -> {
+                res.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                res.setContentType("plain/text");
+                writer.write(e.getMessage());
+            }
+            case (HttpServletResponse.SC_METHOD_NOT_ALLOWED) -> {
+                res.setStatus(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+                res.setHeader("Allow", "POST");
+                res.setContentType("text/plain");
+                writer.write(e.getMessage());
+            }
+        }
+    }
```

Comment:
I'd imagined we'd just need to set the status code based on the exception here, but I can see that you're handling a few different cases here. We should check this again.